### PR TITLE
Document Parquet format for changefeeds

### PR DESCRIPTION
**Pull request by kathancox**
Fixes DOC-6799

Added the parquet format to the changefeed pages.

Also some clean up of missing information on formats in different areas of the docs.

### CODE REVIEW

**Comment by jayshrivastava** [comment]
@kathancox Is it alright if I hold off on reviewing this until https://github.com/cockroachdb/cockroach/issues/99028 is completed? There's a few gaps I need to address still.

**Comment by kathancox** [comment]
@jayshrivastava Totally, whenever you think is best 👍

**Comment by kathancox** [comment]
@jayshrivastava Is the Parquet eng work in a place, where reviewing this docs PR makes sense now?

**Comment by jayshrivastava** [comment]
@kathancox Yup. I'll take a look!

File: src/current/v23.1/changefeed-for.md [hunk 1]
```diff
@@ -71,7 +71,7 @@ Option | Value | Description
...
 `envelope` | `key_only` / `row` / `wrapped` | `key_only` emits only the key and no value, which is faster if you only want to know when the key changes.<br><br>`row` emits the row without any additional metadata fields in the message. `row` does not support [`avro` format](#format).<br><br>`wrapped` emits the full message including any metadata fields. See [Responses](changefeed-messages.html#responses) for more detail on message format.<br><br>Default: `envelope=wrapped`
-<a name="format"></a>`format` | `json` / `avro` | Format of the emitted record. Currently, support for [Avro is limited](changefeed-messages.html#avro-limitations). <br><br>Default: `format=json`.
+<a name="format"></a>`format` | `json` / `avro` / `csv` / `parquet` | Format of the emitted message. <br><br>`avro`: `confluent_schema_registry` is required with this format. Support for [Avro is limited](#avro-limitations). <br><br>`csv`: [`initial_scan=only`](#initial-scan) is required with this format.<br><br>Default: `format=json`.
```
Comment (jayshrivastava) [inline src/current/v23.1/changefeed-for.md:74]: We should mention that parquet only works with the cloudstorage format.

**Comment by kathancox** [comment]
I am closing this PR as the performance related changes to Parquet are not being backported to v23.1. This work will need to go into v23.2.

**Pull request closed without merging**